lib/python/rose/apps/rose_ana.py

```diff
@@ -85,32 +85,53 @@ def create(self):
             status TEXT,
             comparison TEXT,
             PRIMARY KEY(app_task))
-            """)
-        # And this one stores the completion status of the task
-        conn.execute(
             """
+        self.execute_sql_retry(conn, (sql_statement,))
+        # And this one stores the completion status of the task
+        sql_statement = """
             CREATE TABLE IF NOT EXISTS tasks (
             task_name TEXT,
             completed INT,
             PRIMARY KEY(task_name))
-            """)
+            """
+        self.execute_sql_retry(conn, (sql_statement,))
         conn.commit()
 
+    def execute_sql_retry(self, conn, sql_args, retries=10, timeout=5.0):
+        """
+        Given a connection object and a tuple of the desired arguments;
+        calls sql execute and retries multiple times, to handle
+        concurrency issues
+
+        """
+        for retry in range(retries):
+            try:
+                conn.execute(*sql_args)
+                return
+            except:
+                time.sleep(timeout)
+        raise
```

Comment:
A raise on its own like this essentially means "raise whatever the last unhandled exception was"... so in the above case the code will repeatedly retry if an exception occurs, but if it exceeds the number of retries it will hit the raise and whatever the final error was will be re-raised and propagate up the stack (in this case there's nothing to catch it outside so it will bring down the whole app task)